src/plugins/intel_cpu/src/nodes/scaled_attn.cpp

```diff
@@ -388,15 +357,17 @@ struct MHAKernel<ScaledDotProductAttention::KT_ONEDNN, T> {
         size_t h_each_group_len = H / Hk;
         PlainTensor score, weight;
         score.resize({B, H, q_len, kv_len}, static_cast<float*>(attn_score.get_data_handle()));
-        weight.resize({B, H, q_len, kv_len}, static_cast<bfloat16*>(attn_weight.get_data_handle()));
+        weight.resize({B, H, q_len, kv_len}, static_cast<T*>(attn_weight.get_data_handle()));
         const size_t m_block_size = qk_gemm_ptr->get_mblk_size();
         auto m_blocks = (q_len + m_block_size - 1) / m_block_size;
+        bool is_bf16 = precision_of<T>::value == ov::element::bf16;
         // packed k, v
         parallel_for2d(B, Hk, [&](size_t b, size_t h) {
-            bfloat16* k_ptr = &present_key.at<bfloat16>({b, h, 0, 0});
-            bfloat16* v_ptr = &present_value.at<bfloat16>({b, h, 0, 0});
-            qk_gemm_ptr->copy_buffer_b(k_ptr, &qk_scratch_b.at<bfloat16>({b, h, 0}));
-            wv_gemm_ptr->copy_buffer_b(v_ptr, &wv_scratch_b.at<bfloat16>({b, h, 0}));
+            T* k_ptr = &present_key.at<T>({b, h, 0, 0});
+            T* v_ptr = &present_value.at<T>({b, h, 0, 0});
+            qk_gemm_ptr->copy_buffer_b(k_ptr, &qk_scratch_b.at<T>({b, h, 0}));
```

Comment:
just wondering why we need to copy B matrix for fp32 precision? My understanding was that Brgemm can work with B matrix in planar format.

Reply:
Hi Dmitry, the copy_B here is to transpose the Matrix B